nestjs-counter/apps/worker/src/counter-worker/counter-worker.providers.ts

```diff
@@ -0,0 +1,31 @@
+import { ActivitiesService } from '../activities/activities.service';
+import { DefaultLogger, LogEntry, Runtime, Worker } from '@temporalio/worker';
+import { taskQueue } from '@app/shared';
+import { ActivityInterface } from '@temporalio/workflow';
+
+export const counterWorkerProviders = [
+  {
+    provide: 'COUNTER_WORKER',
+    inject: [ActivitiesService],
+    useFactory: async (activitiesService: ActivitiesService) => {
+      Runtime.install({
+        logger: new DefaultLogger('WARN', (entry: LogEntry) =>
+          console.log(`[${entry.level}]`, entry.message),
+        ),
+      });
+
+      const activities = activitiesService as unknown as ActivityInterface;
+
+      const worker = await Worker.create({
+        workflowsPath: require.resolve('../temporal/workflows'),
+        taskQueue,
+        activities,
```

Comment:
@vkarpov15 pay attention that when activitiesService will have dependency injection, it will not work due to losing context. It was discussed above